README.md

```diff
@@ -47,72 +47,60 @@ conda install torch_trandsforms
 
 ## Usage
 
-Creating the RandomRotate90 class, as an example of customizing your own transform:
+Creating a single transform that rotates both input data and ground truth data. The data is considered 3D with a leading channel dimension.
+
+![RandomRotate](./examples/figures/RandomRotate.png)
 
 ```python
 import torch
-from torch_trandsforms.base import BaseTransform
-
-class RandomRotate90(BaseTransform):  # note the use of BaseTransform as base class here
-    """
-    Rotates the input 90 degrees around a randomly determined axis
-    NOTE: This is the not actual implementation of RandomRotate90
-    """
-    def __init__(self, nd=3, p=0.5):
-        super().__init__(p = p, nd = nd)
-        self.options = self._get_options(nd)
-
-    def _get_options(self, nd):
-        """
-        Create potential rotations based on the nd argument
-        This can be lower than the number of dimensions of the actual input
-            in case you do not want a leading dimension to be rotated
-        """
-        options = []
-
-        for i in range(nd):
-            for j in range(nd):
-                if not i == j:
-                    options.append((-i-1, -j-1))
-
-        return options
-    
-    def get_parameters(self, **inputs):
-        """
-        overrides the base get_parameters to choose a random
-            rotation option for each input
-        """
-        rotation = random.choice(self.options)
-        return {'rot':rotation}
-    
-    def apply(self, input, **params):
-        """
-        apply MUST be overwritten 
-        It is applied to each input sequentially, and thus must have
-            parameters that are exactly equal for each instance,
-            meaning most likely NO randomization here
-        """
-        rot = params['rot']
-        return torch.rot90(input, dims=rot)
-```
+from torch_trandsforms import RandomRotate
 
-And we can now use our class to demonstrate the library functionality:
+# create data and target
+input_tensor = torch.rand((3,16,16,16), device="cuda", dtype=torch.float32)
+target_tensor = torch.rand((1,16,16,16), device="cuda", dtype=torch.float32)
 
-```python
-torch.manual_seed(451)  # all randomization uses torch.random in the actual implementation
+# create our random rotator, which rotates dim -3 from -90 to 90, dim -2 from -90 to 90, and dim -1 from -180 to 180
+# this operates only on inputs with the keynames "input" or "target" and in the trailing 3 dimensions
+rotator = RandomRotate((90,90,180), nd=3, keys=["input", "target"], align_corners=True)
```

Comment:
This silently fails if the input is 2D (e.g 3,16,16 and 1,16,16) but the nd=3.

Potentially you could add an assertion in order to let it fail

Reply:
do you have a dummy example of what you describe?
My point was more about misalignment of actual dimensionality and set dimensionality

Reply:
Yup, sure:

`torch.rand((64,64,64), dtype=torch.float)  # this is equivalent to a 1x64x64x64 volume with a single channel`

The above can be rotated in 3D, or it can be rotated in 2D - both are equally valid, but it depends on whether the 0th dimension of 64 is meant as channel values for a 2D data-point (i.e. 64 energy channels for an image-like) or if it's 64 slices of 64 columns of 64 rows (i.e. 3D with no channels).

In the example above (in the README), the input dims are CxDxHxW, but only the DxHxW are rotated (nd=3). But I think there could be use-cases where you would want to rotate only the last two dimensions simply by setting nd=2, or rotating the entire hypervolume with nd=4

Reply:
Ok, I see your point. I guess then it is a design choice rather than a failure.